auto_editor/AddCommentsRule.py

```diff
@@ -39,17 +39,44 @@ def run_rule(self, project: StructuredProjectSource,
                 state = False
 
                 # add if statement for the corresponding warning type, and change the recommendation
+                if w_type == "DPCT1065":
+                    del all_items[warning_first_line - 1: warning_last_line]
+                    comment_item = LineItem("/*CTA1065:count number: recommended to ignore this warning. but you can also consider replacing 'item_ct1.barrier();' with 'item_ct1.barrier(sycl::access::fence_space::local_space);'*/\n")
+                    all_items.insert(warning_first_line - 1, comment_item)
+                    w_type = "DPCT"
+                    return project
 
-                if w_type == "DPCT1049":
+                if w_type == "DPCT1039":
                     del all_items[warning_first_line - 1: warning_last_line]
-                    comment_item = LineItem("/*CTA1049:count number: recommendation*/\n")
+                    comment_item = LineItem("/*CTA1039:count number: Base on the experience, strongly recommended to leave the code as it is and ignore this warning. BUT, if '&xxx' points to local memory, please accept the DPCT proposal.*/\n")
                     all_items.insert(warning_first_line - 1, comment_item)
                     w_type = "DPCT"
                     return project
 
-                if w_type == "DPCT1065":
+                if w_type == "DPCT1008":
                     del all_items[warning_first_line - 1: warning_last_line]
-                    comment_item = LineItem("/*CTA1065:count number: recommended to ignore this warning. but you can also consider replacing 'item_ct1.barrier();' with 'item_ct1.barrier(sycl::access::fence_space::local_space);'*/\n")
```

Comment:
it would be better to count indentation spaces there.
 def get_indentation_spaces(self,new_code):
        j, prefix = 0, ""
        while j < len(new_code):
            if new_code[j] == " ":
                prefix = prefix + " "
                j += 1
            else:
                break
        return prefix

这个功能就是计算一下这个提示前的空格个数，让提示和代码对齐。